### PR TITLE
Python: port code injection query

### PR DESCRIPTION
**Pull request by RasmusWL**
One thing for discussion:

Instead of implementing a generic additional taint step for `compile`, I considered to make it query specific. I don't _really_ have any strong arguments for or against, so would love input on whether you agree on this decision @tausbn and @yoff. (it enabled me to test it as part of the framework tests instead of the tests for the relevant query, but I wouldn't call that a strong argument.)

### CODE REVIEW

File: python/ql/src/experimental/Security-new-dataflow/CWE-094/CodeInjection.ql [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * @name Code injection
+ * @description Interpreting unsanitized user input as code allows a malicious user arbitrary
```
Comment (intrigus-lgtm) [inline python/ql/src/experimental/Security-new-dataflow/CWE-094/CodeInjection.ql:3]:
```suggestion
 * @description Interpreting unsanitized user input as code allows a malicious user to perform arbitrary
```

Comment (RasmusWL) [reply python/ql/src/experimental/Security-new-dataflow/CWE-094/CodeInjection.ql:3]: Thanks :+1:

**Review by yoff — DISMISSED** [review]
Looks good

**Comment by yoff** [comment]
On the topic of `compile`: As I understand it, one currently simply has to include `StdLib` (for example by including `Frameworks`) and then `compile` will preserve taint. I think that sounds great.

**Comment by RasmusWL** [comment]
> On the topic of `compile`: As I understand it, one currently simply has to include `StdLib` (for example by including `Frameworks`) and then `compile` will preserve taint. I think that sounds great.

Nice :+1: (note that the new file is actually called `Stdlib.qll`, following convention from either JS or Go -- can't quite recall)

@tausbn what do you think about this maybe-only-query-specific additional taint step that I made a globally applicable taint step? (question outlined in PR description) -- going to add you as reviewer explicitly to get your thoughts on this one.

**Review by yoff — APPROVED** [review]
LGTM

**Comment by RasmusWL** [comment]
Had to manually solve merge-conflict (since multiple new concepts were added). Sadly, that means old review was discarded :neutral_face:

**Review by yoff — APPROVED** [review]
Condensed view of merge looks ok, full one is scary...

**Review by tausbn — APPROVED** [review]
I think adding `compile` as a generic taint-preserving step is fine. Certainly, I would expect anything coming out of a user-controlled compilation to be very sus.

**Pull request merged**